src/components/App.tsx

```diff
@@ -136,6 +136,7 @@ let isHoldingSpace: boolean = false;
 let isPanning: boolean = false;
 let isDraggingScrollBar: boolean = false;
 let currentScrollBars: ScrollBars = { horizontal: null, vertical: null };
+const resizeWithCenter = { keyLifted: false };
```

Comment:
Are you sure we need a mutable object here?
(BTW, is it somewhat related with #1340?)

Reply:
We need whether a mutable object or `resizeElements` returning some flag to App in order to update the value.

#1340 will be different implementation as we will have to call resizeElements onKeyDown / Up, I think...

Reply:
Sorry, I should have described my question more clearly. One question, and one comment.

1. I would like to understand the purpose of this flag for the future. Why does it require a flag in App? Not like `const lifted = getResizeCenterPointKey(event);` inside resizeElements.ts?
2. If we need to mutate object in App, I would like to have a consistent way. When I refactored resizeElements for the first time, I used `setAppState` and `setResizeArrowFn`, which I thought cleaner. But, it may be just a preference, and I'm OK to change that. If we take the mutable object approach, I would like to change `setResizeArrowFn` too. We can do this after this PR is merged anyway.

Reply:
Hey @dai-shi , I'll try to explain my point:

1. `resizeWithCenter.keyLifted` flag is needed because that variable will be shared between `onKeyDown` and `resizeElements`. So, when the user stop pressing the key alt, then there is no need for the user to move the mouse to have the changes applied in the scene. If we use `const lifted = getResizeCenterPointKey(event);` then we can not achieve that behaviour because resizeElements is called when mouse move.

2. I see your approach now. I will follow the current approach and remove the mutable object.

Reply:
Hey @jquintozamora , thanks for the explanation. And, 1) that sounds exactly like the problem of #1340. So, if I understand it correctly, #1340 can be solved with the same technique, and in that case there _could_ be common/similar logic/pattern, or maybe not.

Reply:
I think #1340 could be solved in a similar way, but we will have to call `resizeElements` inside `onKeyDown` in order to directly apply the new size somehow. I think :)

Reply:
Hm, that part I didn't understand. I will probably know it when I look around this logic closely in the possible future. Thanks again for your explanation.